src/wrappers/digift/DigiftWrapper.sol

```diff
@@ -0,0 +1,1118 @@
+     * @param shares The number of shares to mint (must match maxMint)
+     * @param receiver The address to receive the shares
+     * @param controller The controller address (must be msg.sender)
+     * @return assets The amount of assets used for the deposit
+     */
+    function mint(uint256 shares, address receiver, address controller)
+        public
+        onlyWhitelistedNode
+        nonReentrant
+        returns (uint256 assets)
+    {
+        _actionValidation(shares, controller, receiver);
+        NodeState storage node = _nodeState[msg.sender];
+
+        // Ensure deposit request has been fulfilled and exact shares are minted
+        require(node.claimableDepositRequest > 0, DepositRequestNotFulfilled());
+        require(node.maxMint == shares, MintAllSharesOnly());
+
+        assets = node.claimableDepositRequest;
+        uint256 assetsToReimburse = node.pendingDepositReimbursement;
+
+        // Clear node state
+        node.claimableDepositRequest = 0;
+        node.maxMint = 0;
+        node.pendingDepositReimbursement = 0;
+
+        // Mint shares to the node
+        _mint(msg.sender, shares);
+
+        // Reimburse unused assets to the node
+        if (assetsToReimburse > 0) {
+            IERC20(asset).safeTransfer(msg.sender, assetsToReimburse);
+        }
+
+        emit Deposit(controller, receiver, assets - assetsToReimburse, shares);
+    }
+
+    // =============================
+    //     Redemption Functions
+    // =============================
+
+    /**
+     * @notice Request a redemption of shares from a whitelisted node
+     * @dev Transfers shares from node to contract and tracks the request
+     * @param shares The number of shares to redeem
+     * @param controller The controller address (must be msg.sender)
+     * @param owner The owner address (must be msg.sender)
+     * @return The request ID (always 0 in this implementation)
+     */
+    function requestRedeem(uint256 shares, address controller, address owner)
+        external
+        onlyWhitelistedNode
+        nonReentrant
+        returns (uint256)
+    {
+        _actionValidation(shares, controller, owner);
+        _nothingPending();
+
+        _nodeState[msg.sender].pendingRedeemRequest = shares;
+        _globalState.accumulatedRedemption += shares;
+
+        _spendAllowance(msg.sender, address(this), shares);
+        _transfer(msg.sender, address(this), shares);
+
+        emit RedeemRequest(controller, owner, REQUEST_ID, msg.sender, shares);
+        return REQUEST_ID;
+    }
+
+    /**
+     * @notice Settle redemption requests for multiple nodes
+     * @dev Verifies Digift redemption event and distributes assets/shares proportionally
+     * @param nodes Array of node addresses to settle
+     * @param verifyArgs Offchain arguments for event verification
+     */
+    function settleRedeem(address[] calldata nodes, DigiftEventVerifier.OffchainArgs calldata verifyArgs)
```

Comment:
I acknowledge that there might be a potential DDoS scenario resulting in an out-of-gas error if too many nodes are involved. Realistically, the number of nodes would have to be extremely large for that to occur. Additionally, the manager of this contract can call forwardRequestsToDigift more frequently to start processing requests (for example, handling 100 nodes at once).